### DISCUSSION TITLE
Any support for dacpac refactor log?

### DISCUSSION THREAD

**Discussion by asrichesson** [microsoft/DacFx#289] [Ideas] [answered]
I know in the old project format you could include a refactor log for storing object renames as discussed here:

https://learn.microsoft.com/en-us/sql/ssdt/how-to-use-rename-and-refactoring-to-make-changes-to-your-database-objects?view=sql-server-ver16

These refactors would be baked into the dacpac and applied (if necessary) on deployment. 

Obviously, there isn't any visual studio support yet, but is adding the refactor log entries manually an option?

I would expect it to look something like this in the csproj file:
```xml
<RefactorLog Include="MyDatabase.refactorlog" />
```
And the refactor log:
```xml
<?xml version="1.0" encoding="utf-8"?>
<Operations Version="1.0" xmlns="http://schemas.microsoft.com/sqlserver/dac/Serialization/2012/02">
  <Operation Name="Rename Refactor" Key="1f43734f-1598-4c41-a2f5-1ca07a32cf45" ChangeDateTime="03/04/2023 00:44:44">
    <Property Name="ElementName" Value="[dbo].[MyTable].[MyColumn]" />
    <Property Name="ElementType" Value="SqlSimpleColumn" />
    <Property Name="ParentElementName" Value="[dbo].[MyTable]" />
    <Property Name="ParentElementType" Value="SqlTable" />
    <Property Name="NewName" Value="[MyNewColumnName]" />
  </Operation>
</Operations>
```

**Comment by lavbnon** [comment]
Same question for me, did not find anything yet to support changes in table structures.

**Comment by asrichesson** [comment, marked as answer]
Adding this to your sdk sqlproj seems to work.
```xml
<ItemGroup>
  <RefactorLog Include="SdkTestDatabase.refactorlog" />
</ItemGroup>
```

**Comment by EklipZgit** [comment]
Does not appear to be working for me 🤷

**Comment by dzsquared** [comment]
Tracking finishing the documentation on refactorlog here - https://github.com/microsoft/DacFx/issues/770

It should/does work, but is not currently documented enough that I expect you to be able to guess how it works.
